manager/host/skyline.go

```diff
@@ -0,0 +1,239 @@
+	IdcProvince        string   `json:"idc.province"`
+	IdcCity            string   `json:"idc.city"`
+	RoomAbbreviation   string   `json:"room.abbreviation"`
+	CabinetCabinetNum  string   `json:"cabinet.cabinet_num"`
+	CabinetLogicRegion string   `json:"cabinet.logic_region"`
+	DswClusterName     string   `json:"dsw_cluster.name"`
+	NetAswName         string   `json:"net_asw.name"`
+	LogicPodName       string   `json:"logic_pod.name"`
+	NetPodName         string   `json:"net_pod.name"`
+	Tags               []string `json:"tags"`
+}
+
+type Value struct {
+	TotalCount int     `json:"totalCount"`
+	HasMore    bool    `json:"hasMore"`
+	ItemList   []*Item `json:"itemList"`
+}
+
+type QueryResponse struct {
+	Success      bool   `json:"success"`
+	Value        *Value `json:"value"`
+	ErrorCode    int    `json:"errorCode"`
+	ErrorCodeMsg string `json:"errorCodeMsg"`
+	ErrorMessage string `json:"errorMessage"`
+	AllowRetry   bool   `json:"allowRetry"`
+}
+
+type Auth struct {
+	AppName   string `json:"appName"`
+	Account   string `json:"account"`
+	Timestamp int64  `json:"timestamp"`
+	Signature string `json:"signature"`
+}
+
+type skylineClient struct {
+	domain    string
+	appName   string
+	account   string
+	accessKey string
+}
+
+type ItemSearchRequest struct {
+	Auth *Auth         `json:"auth"`
+	Item *QueryRequest `json:"item"`
+}
+
+func (client *skylineClient) getSignature(timestamp string) string {
+	str := client.account + client.accessKey + timestamp
+	h := md5.New()
+	h.Write([]byte(str))
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func (client *skylineClient) itemSearch(req *QueryRequest) (*QueryResponse, error) {
+	now := time.Now().Unix()
+	body, err := json.Marshal(&ItemSearchRequest{
+		Auth: &Auth{
+			AppName:   client.appName,
+			Account:   client.account,
+			Timestamp: now,
+			Signature: client.getSignature(strconv.FormatInt(now, 10)),
+		},
+		Item: req,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	url := fmt.Sprintf("%s/item/query", client.domain)
+	request, err := http.NewRequest("POST", url, bytes.NewReader(body))
+	defer request.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Set("Content-Type", "application/json;charset=UTF-8")
+	request.Header.Set("account", client.account)
+
+	cli := &http.Client{}
+	response, err := cli.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	ret, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &QueryResponse{}
+	err = json.Unmarshal(ret, resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+type skyline struct {
+	client *skylineClient
+}
+
+func NewSkyline(config *config.SkylineService) (HostManager, error) {
+	return &skyline{
+		client: &skylineClient{
+			domain:    config.Domain,
+			appName:   config.AppName,
+			account:   config.Account,
+			accessKey: config.AccessKey,
+		}}, nil
+}
+
+var SelectLists = []string{
+	"ip", "sn", "tags", "host_name", "security_domain", "vpc_id",
```

Comment:
Use lower camel case.

Reply:
bind with skyline sql backend, these are key words of query